aten/src/ATen/native/cuda/Blas.cu

```diff
@@ -4,35 +4,59 @@
 
 namespace at { namespace native {
 
-Tensor &addmv_impl_cuda(Tensor& result, const Tensor &self, const Tensor &mat, const Tensor &vec, const Scalar& beta_, const Scalar& alpha_) {
-  auto r_stride = result.stride(0);
-  auto vec_stride = vec.stride(0);
-
-  // Check for contiguity of `vec` and update `vec_stride` accordingly
-  const auto vec_contiguous = vec_stride == 0 ? vec.contiguous() : vec;
-  vec_stride = vec_contiguous.stride(0);
-
-  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, mat.scalar_type(), "addmv_impl_cuda", [&] {
-    auto beta = beta_.to<scalar_t>();
-    auto alpha = alpha_.to<scalar_t>();
-    if (mat.stride(0) == 1 && mat.stride(1) >= std::max<int64_t>(1, mat.size(0))) {
-      at::cuda::blas::gemv<scalar_t>('n',
-        mat.size(0), mat.size(1), alpha, mat.data_ptr<scalar_t>(), mat.stride(1), vec_contiguous.data_ptr<scalar_t>(),
-        vec_stride, beta, result.data_ptr<scalar_t>(), r_stride);
+TORCH_IMPL_FUNC(addmv_out_cuda)(const Tensor &self, const Tensor &mat, const Tensor &vec, const Scalar& beta_, const Scalar& alpha_, const Tensor& result) {
+  Tensor self_ = self;
+  if (self.numel()==1) {
+    self_ = self.expand({mat.size(0)});
+  }
+  auto betaval = beta_.toComplexDouble();
+  if (mat.numel() == 0) {
+    // shortcut for an empty matrix
+    // By definition, when beta==0, values in self should be ignored. nans and infs
+    // should not propagate
+    if (betaval == 0.0) {
+      result.zero_();
+    } else {
+      at::mul_out(
```

Comment:
neither at::native::mul_out nor at::cpu::mul_out work here, so another dispatch it is

Reply:
what about `at::cuda::mul_out` 8)

Reply:
`"at::cuda" has no member "mul_out"` - do I need some includes?